src/main/java/com/neilbaner/duke/serialize/Serializer.java

```diff
@@ -0,0 +1,121 @@
+ * The last two tokens represent the task title, and the "done" status
+ * (as Y/N for done/not done) respectively.
+ * @author neilbaner
+ */
+public class Serializer {
+    /**
+     * Takes in an object of type TaskList and returns a String
+     * which can then be written to a file or used otherwise.
+     * @param tl TaskList that we want to serialize.
+     * @return String containing serialized output
+     */
+    public static String serializeTaskList(TaskList tl) {
+        String serialized = "";
+        for (Task t : tl.getAllTasksList()) {
+            serialized += serializeTask(t);
+            serialized += "~~";
+        }
+        serialized = serialized.substring(0, serialized.length() - 2);
+        return serialized;
+    }
+
+    /**
+     * Takes a String created by serializeTaskList() and deserializes it
+     * into an ArrayList&lt;Task&rt;.
+     * @param s The String (eg. from a text file) for the entire task list
+     * @return ArrayList&lt;Task&gt; containing all the Task objects represented in the String
+     * @throws DeserializerException if there is an error deserializing any task
+     */
+    public static ArrayList<Task> deserializeTaskList(String s) throws DeserializerException {
+        ArrayList<Task> taskList = new ArrayList<Task>();
+        String[] tasks = s.split("~~");
+        for (int i = 0; i < tasks.length; i++) {
+            taskList.add(deserializeTask(tasks[i]));
+        }
+        return taskList;
+    }
+
+    /**
+     * Takes a single Task object, and converts it into a String
+     * to be written to a file or otherwise stored/transmitted.
+     * @param t the Task object to be serialized
+     * @return String representing the task
+     */
+    public static String serializeTask(Task t) {
+        Type taskType = t.getClass();
+        String serialized = "";
+        if (taskType == ToDo.class) {
+            serialized += "T~";
+        } else if (taskType == Event.class) {
+            serialized += "E~";
+            serialized += ((Event) t).getEventTime() + "~";
+        } else if (taskType == Deadline.class) {
+            serialized += "D~";
+            serialized += ((Deadline) t).getDueDate() + "~";
+        }
+        serialized += t.getTitle() + "~";
+        serialized += (t.getDone()) ? "Y" : "N";
+        return serialized;
+    }
+
+    /**
+     * Takes a String generated by serializeTask() and attempts to deserialize it
+     * into a single Task object (will be of correct subclass type, i.e. ToDo, Deadline, Event)
+     * @param s A string, generated by serializeTask() to deserialize
+     * @return Task subclass (ToDo, Event, Deadline) as represented by s
+     * @throws DeserializerException if there is some error in the string.
+     */
+    public static Task deserializeTask(String s) throws DeserializerException {
+        String parts[];
+        Task deserialized;
+        parts = s.split("~");
+        try {
+            if (parts[0].equals("T")) {
```

Comment:
please try to have a variable name for each of the array element so that its easier when you refer to it in the future.